src/bluetooth/DataAccess.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2021
+ */
+
+/**
+ * Helper methods to perform data access functionality for breach calculations
+ */
+import VACCINE_ENTITIES from '../utilities/modules/vaccines/constants';
+
+export class DataAccess {
+  constructor(dbService) {
+    this.db = dbService;
+  }
+
+  getTemperatureLogsFrom = (sensorId, timeToCheckFrom) =>
+    this.db
+      .objects(VACCINE_ENTITIES.TEMPERATURE_LOG)
+      .filtered('sensor.id == $0 AND timestamp >= $1', sensorId, timeToCheckFrom)
+      .sorted('timestamp');
+
+  getMostRecentBreachLog = sensorId => {
+    const allBreachLogs = this.db
+      .objects(VACCINE_ENTITIES.TEMPERATURE_LOG)
+      .filtered('sensor.id == $0', sensorId);
+    const [mostRecentBreachLog] = allBreachLogs.sorted('timestamp', true);
+
+    return mostRecentBreachLog;
+  };
+
+  getMostRecentBreach = sensorId => {
+    const allBreaches = this.db
+      .objects(VACCINE_ENTITIES.TEMPERATURE_BREACH)
+      .filtered('sensorId == $0', sensorId);
+    const [mostRecentBreach] = allBreaches.sorted('startTimestamp', true);
+
+    return mostRecentBreach;
+  };
+
+  getBreachConfigs = () => {
+    const configs = this.db
+      .objects(VACCINE_ENTITIES.TEMPERATURE_BREACH_CONFIGURATION)
+      .filtered('id == "HOT_BREACH" OR id == "COLD_BREACH"');
```

Comment:
We're gonna use `type` field now just to mix it up. The types will be `HOT_CUMULATIVE`, `HOT_CONSECUTIVE`, `COLD_CUMULATIVE` and `COLD_CONSECUTIVE`. This will need to return the `_CONSECUTIVE` ones

Reply:
Oh right that makes sense, I see it in the updated schema too - updated with 4aaf415b